library/control/src/modules/ProductControl.rb

```diff
@@ -1591,11 +1601,29 @@ def ResetAdditionalWorkflowParams
       nil
     end
 
+    # Add new system roles
+    #
+    # For the time being, roles are inserted at the beginning of the
+    # roles list.
+    #
+    # @example Adding a simple role
+    #   ProductControl.system_roles #=> []
+    #   ProductControl.add_system_roles([{"id" => "new_role"}]) #=> [{"id" => "new_role"}]
+    #   ProductControl.system_roles #=> [{"id" => "new_role"}]
+    #
+    # @param [Array<Hash>] new_roles Roles to add
+    # @return [Array<Hash>] All existing roles after adding the new roles
```

Comment:
But the caller does not use that, and the `publish` declaration says `void` too.